z stream 2.14.1 release note (#8281)
open-cluster-management/rhacm-docs: release_notes/acm_fixed_issues.adoc

link:https://issues.redhat.com/browse/ACM-24849[ACM-24849]

* Updates the console with a functional *Upgrade* button to start the process with accurate version upgrade availability.
link:https://issues.redhat.com/browse/ACM-23943[ACM-23943]

* Fixes pull model GitOps deployments that were stuck in a constant `Refreshing` state in the console to a stable state. The `pull-integration-controller` clears the refresh annotation to prevent Applications from getting stuck in a constant `Refreshing` status.
link:https://issues.redhat.com/browse/ACM-23908[ACM-23908]

* Fixes a formatting issue in the assisted service validation function that caused failure from a mismatch between the cluster machine CIDR and the calculated CIDR during ipv6 installation. The fix ensures that the CIDR values are compared correctly.
link:https://issues.redhat.com/browse/ACM-23289[ACM-23289]

* Reduces the excessive `Skipping this secret` logs in the `certificate-policy-controller` resource by logging parsing errors and moving logs about secrets without a certificate.

> link:https://issues.redhat.com/browse/ACM-23248[ACM-23248]

Reviewer: LGTM!

* Fixes a race condition where a host that was added to a managed cluster was marked as `installed`, `Done` by Central Infrastructure Management before all of the Certificate Signing Requests were approved. Central Infrastructure Management now approves all Certificate Signing Requests for hosts added to a managed cluster before marking the host as completely installed (Done).

> link:https://issues.redhat.com/browse/ACM-22786[ACM-22786]

Reviewer 2: lgtm

Thanks Brandi!

* Fixes when a namespace containing a Go template string but no `namespaceSelector` was incorrectly determined to not have a namespace that is specified. The fix ensures that the namespace is correctly identified.

> link:https://issues.redhat.com/browse/ACM-22702[ACM-22702]

Reviewer: LGTM!

* Fixes error where the `objectSelector` and `namespaceSelector` were used at the same time, and objects that were discovered by the `objectSelector` in specific namespaces were applied to all namespaces that were returned by the namespaceSelector. 

> link:https://issues.redhat.com/browse/ACM-22676[ACM-22676]

Reviewer: LGTM!

*Important:* Cluster lifecycle components and features are within the {mce-short}, which is a software operator that enhances cluster fleet management. Release notes for {mce-short}-specific features are found in at link:../clusters/release_notes/mce_release_notes.adoc#mce-release-notes[Release notes for Cluster lifecycle with {mce-short}].

== Errata 2.14.1

* Delivers updates to one or more product container images.

> * Adds a missing link to the operator source code and repository on the OperatorHub installation page. The 'annotation.repository' is updated to point to the correct repository URL.

Reviewer 3: same as previous suggestions. But I see in the linked issue that MCE is only mentioned. Should this entry be only for MCE? If yes, remove this line

Author: It's the same fix for both operators, yes. This still has to get reviewed by dev, but your feedback is correct. TY!

> * Adds a missing link to the operator source code and repository on the OperatorHub installation page. The 'annotation.repository' is updated to point to the correct repository URL.

Reviewer 3: ```suggestion
* Adds a missing link to the operator source code and repository on the OperatorHub installation page. The `annotation.repository` is updated to point to the correct repository URL.
```

link:https://issues.redhat.com/browse/ACM-24849[ACM-24849]

* Updates the console with a functional *Upgrade* button to start the process with the accurate version for upgrade availability.
link:https://issues.redhat.com/browse/ACM-23943[ACM-23943]

> * Fixes pull model GitOps deployments that were stuck in a constant `Refreshing` state in the console to a `Stable` state. The `pull-integration-controller` clears the refresh annotation to prevent Applications from getting stuck in a constant `Refreshing` status.

Reviewer 3: When i searched for "pull model GitOps", nothing appeared in the results. However, when I searched for pull model, I saw "Argo CD pull model. Should it read as, "Fixes `GitOps` deployments in the Argo CD pull model that were stuck in a constant...."

```suggestion
* Fixes pull model GitOps deployments that were stuck in a constant `Refreshing` state in the console to a `Stable` state. The `pull-integration-controller` clears the refresh annotation to prevent applications from getting stuck in a constant `Refreshing` status.
```

Reviewer 3: @swopebe

link:https://issues.redhat.com/browse/ACM-23908[ACM-23908]

> * Updates validation logic to normalize user-provided CIDRs before comparison with the calculate one, which was a formatting issue in the assisted service validation function that caused failure from a mismatch between the cluster machine CIDR and the calculated CIDR during IPv6 installation. The validation failed when users provided uppercase letters, leading zeros, or both.

Reviewer 3: Run on sentence here

```suggestion
* Updates validation logic to normalize user-provided CIDRs before comparing with the calculated one, which was a formatting issue in the assisted service validation function. The formatting issue caused a failure from a mismatch between the cluster machine CIDR and the calculated CIDR during IPv6 installation. The validation failed when users provided uppercase letters, leading zeros, or both.
```

Author: hhhmmm: _Updates validation logic to normalize user-provided CIDRs before comparing with the calculated one, which was a formatting issue_ 

-- not really correct grammar, even before-- now that I look at it. The subject and predicate don't agree here. I will work on it a bit.

Author: Changed this a bit.

Author: This was rewritten outside of this comment.

> * Updates validation logic to normalize user-provided Classless Inter-Domain Routing (CIDR) before comparison with the calculate CIDR to prevent a formatting issue in the assisted service validation function. A mismatch between the cluster machine CIDR and the calculated CIDR during IPv6 installation led to failure when users provided uppercase letters, leading zeros, or both.

Reviewer 3: ```suggestion
* Updates validation logic to normalize user-provided Classless Inter-Domain Routing (CIDR) before comparison with the calculated CIDR to prevent a formatting issue in the assisted service validation function. A mismatch between the cluster machine CIDR and the calculated CIDR during IPv6 installation led to failure when users provided uppercase letters, leading zeros, or both.
```

link:https://issues.redhat.com/browse/ACM-23289[ACM-23289]

* Reduces the excessive `Skipping this secret` logs in the `certificate-policy-controller` resource by logging parsing errors and moving logs about secrets without a certificate.
link:https://issues.redhat.com/browse/ACM-23248[ACM-23248]

> * Fixes a race condition where a host that was added to a managed cluster was marked as `installed`, `Done` by {cim} before all of the Certificate Signing Requests were approved. {cim} now approves all Certificate Signing Requests for hosts that are added to a managed cluster before marking the host as completely `installed` and `Done`.

Reviewer 3: Reminder that {cim} uses lowercase for the operator name (central infrastructure management)

> * Fixes a race condition where a host that was added to a managed cluster was marked as `installed`, `Done` by {cim} before all of the Certificate Signing Requests were approved. {cim} now approves all Certificate Signing Requests for hosts that are added to a managed cluster before marking the host as completely `installed` and `Done`.

Author: ```suggestion
* Fixes a race condition where a host that was added to a managed cluster was marked as `installed`, `Done` by {cim} before all of the Certificate Signing Requests were approved. The {cim} now approves all Certificate Signing Requests for hosts that are added to a managed cluster before marking the host as completely `installed` and `Done`.
```

> * Fixes a race condition where a host that was added to a managed cluster was marked as `installed`, `Done` by Central Infrastructure Management before all of the Certificate Signing Requests were approved. Central Infrastructure Management now approves all Certificate Signing Requests for hosts added to a managed cluster before marking the host as completely installed (Done).

Reviewer 3: ```suggestion
* Fixes a race condition where a host that was added to a managed cluster was marked as `installed`, `Done` by {cim} before all of the Certificate Signing Requests were approved. The {cim} now approves all Certificate Signing Requests for hosts added to a managed cluster before marking the host as completely installed.
```

Author: I think we need to keep `Done` here. Just not in parentheses. I will change that.

Author: The line was rewritten outside of this comment.

link:https://issues.redhat.com/browse/ACM-22786[ACM-22786]

* Fixes when a namespace containing a Go template string but no `namespaceSelector` was incorrectly determined to not have a namespace that is specified. The fix ensures that the namespace is correctly identified.
link:https://issues.redhat.com/browse/ACM-22702[ACM-22702]

> * Fixes error where the `objectSelector` and `namespaceSelector` were used at the same time, and objects that were discovered by the `objectSelector` in specific namespaces were applied to all namespaces that were returned by the namespaceSelector. 

Reviewer 3: ```suggestion
* Fixes error where the `objectSelector` and `namespaceSelector` were used at the same time, and objects that were discovered by the `objectSelector` in specific namespaces were applied to all namespaces that were returned by the `namespaceSelector`. 
```

link:https://issues.redhat.com/browse/ACM-22676[ACM-22676]

> * Corrects a race condition in the Observability `metric-collector`,  occurred when you used multiple workers. This caused some metrics to be incorrectly collected, causing gaps in the metrics data. 

Reviewer 3: there's extra space before "occurred". A suggested the word "creating" since "caused" was already used in the sentence.
```suggestion
* Corrects a race condition in the Observability `metric-collector` that occurred when you used multiple workers. This caused some metrics to be incorrectly collected, creating gaps in the metrics data. 
```

> * Corrects a race condition in the Observability `metric-collector`, which occurred when you used multiple workers. This caused some metrics to be incorrectly collected, causing gaps in the metrics data. 

Reviewer 4: LGTM, thanks!

> * Corrects the faulty metric collection from Prometheus. When using multiple workers with race condition, some worker nodes collected the same metrics instead of distinct sets of metrics, and sent metrics to the hub cluster Observability configuration.

Reviewer 3: im a little confused on this entry. "When using multiple workers with race condition.." 

I took a look at the summary from the issue and suggest the following edit:

```suggestion
* Corrects the faulty metric collection from Prometheus. A race condition occurred when you used multiple workers. Some worker nodes collected the same metrics instead of distinct sets of metrics, and sent metrics to the hub cluster Observability configuration.
```

Author: I don't think I can say it still occurs; and the summary not clear on this. Unless you are sure you know what race condition means and if it still exists, I am going to pause on that change.

Reviewer 3: Ok sounds good. Race condition is a software vulnerability. I edited the suggestion though

Author: Yeah, that is less risky of losing meaning. This still needs dev review but your new change makes better sense to start with... and very happy to have your knowledge!

> * Corrects the faulty metric collection from Prometheus. A race condition occurred when you used multiple workers. Some worker nodes collected the same metrics instead of distinct sets of metrics, and sent metrics to the hub cluster Observability configuration.

Reviewer 4: There are some correctness issues here. I added two suggestions below - a longer explanation and a more brief one. Perhaps the brief one is enough since we also link to the Jira which has more info.

```suggestion
* Corrects a race condition in the Observability metric-collector which occurred when you used multiple workers. Workers would sometimes collect the same metrics from Prometheus instead of each worker collecting a distinct sets of metrics. This would result in some metrics not being correctly collected and sent to the hub, causing gaps in the metrics data.
```

```suggestion
* Corrects a race condition in the Observability metric-collector which occurred when you used multiple workers. This could cause some metrics to not be correctly collected causing gaps in metrics data. 
```

link:https://issues.redhat.com/browse/ACM-24921[ACM-24921]

> * Fixes the `nmstatectl` binary missing for the s390x architecture, causing the `initrd` image to fail to read the binary. The `nmstatectl` is available and included for the s390x architecture. 

Reviewer 5: ```suggestion
* Adds the `nmstatectl` binary to the initrd image for the s390x architecture which was previously missing.
```

link:https://issues.redhat.com/browse/MGMT-21276[MGMT-21276]

> * Corrects the static network configuration failure to apply on systems with multiple network interfaces that use full ISO, resulting in cluster deployment failure. The static network configuration is applied correctly on systems with multiple network interfaces that use full ISO.

Reviewer 5: Maybe just:

```suggestion
* Corrects a race condition which caused the static network configuration to not apply on systems with many network interfaces that use full ISO.
```

> link:https://issues.redhat.com/browse/MGMT-21688[MGMT-21688]

Author: line 43 and line 46 were generated and edited, please check these. They did not have summaries.
